### ISSUE TITLE
Error reverting / closing buffers when parsing still going on

### ISSUE THREAD

**Issue by zenspider** [zenspider/enhanced-ruby-mode#174]
Related to work being done in #171, I'm seeing this backtrace when I revert a buffer "too quickly" (guess):

```elisp
Debugger entered--Lisp error: (error "Format specifier doesn’t match argument type")
  format("%s%d:%d:%d:%d:%d:" a 4 1 266010 nil nil)
  (process-send-string (erm-ruby-get-process) (format "%s%d:%d:%d:%d:%d:" pc erm-buff-num (point-min) (point-max) min len))
  (if (eq pc 'dbuf) (erm-reparse-diff-buf) (setq erm-parse-buff (current-buffer)) (process-send-string (erm-ruby-get-process) (format "%s%d:%d:%d:%d:%d:" pc erm-buff-num (point-min) (point-max) min len)) (process-send-region erm-ruby-process min max) (process-send-string erm-ruby-process erm-process-delimiter))
  (catch 'interrupted (if (eq pc 'dbuf) (erm-reparse-diff-buf) (setq erm-parse-buff (current-buffer)) (process-send-string (erm-ruby-get-process) (format "%s%d:%d:%d:%d:%d:" pc erm-buff-num (point-min) (point-max) min len)) (process-send-region erm-ruby-process min max) (process-send-string erm-ruby-process erm-process-delimiter)) nil)
  (setq interrupted-p (catch 'interrupted (if (eq pc 'dbuf) (erm-reparse-diff-buf) (setq erm-parse-buff (current-buffer)) (process-send-string (erm-ruby-get-process) (format "%s%d:%d:%d:%d:%d:" pc erm-buff-num (point-min) (point-max) min len)) (process-send-region erm-ruby-process min max) (process-send-string erm-ruby-process erm-process-delimiter)) nil))
  (let ((pc (if erm-parsing-p (if (eq erm-parse-buff (current-buffer)) (setq erm-parsing-p 'a) 'dbuf) (setq erm-response "") (setq erm-parsing-p t) (if (not erm-full-parse-p) (if erm-no-parse-needed-p (progn (setq erm-parsing-p nil) 'a) 'p) (setq min (point-min) max (point-max) len 0 erm-full-parse-p nil) 'r))) interrupted-p) (setq interrupted-p (catch 'interrupted (if (eq pc 'dbuf) (erm-reparse-diff-buf) (setq erm-parse-buff (current-buffer)) (process-send-string (erm-ruby-get-process) (format "%s%d:%d:%d:%d:%d:" pc erm-buff-num (point-min) (point-max) min len)) (process-send-region erm-ruby-process min max) (process-send-string erm-ruby-process erm-process-delimiter)) nil)) (if interrupted-p (progn (setq erm-full-parse-p t))))
  erm-req-parse(nil nil nil)
  (if (and erm-parsing-p (not (eq erm-parse-buff (current-buffer)))) (erm-reparse-diff-buf) (setq erm-full-parse-p t) (erm-req-parse nil nil nil))
  enh-ruby-fontify-buffer()
  (if (enh-ruby-local-enable-extra-keywords) nil (enh-ruby-fontify-buffer))
  erm-reset-buffer()
  (let ((delay-mode-hooks t)) (prog-mode) (setq major-mode 'enh-ruby-mode) (setq mode-name "EnhRuby") (progn (if (get 'prog-mode 'mode-class) (put 'enh-ruby-mode 'mode-class (get 'prog-mode 'mode-class))) (if (keymap-parent enh-ruby-mode-map) nil (set-keymap-parent enh-ruby-mode-map (current-local-map))) (let ((parent (char-table-parent enh-ruby-mode-syntax-table))) (if (and parent (not (eq parent (standard-syntax-table)))) nil (set-char-table-parent enh-ruby-mode-syntax-table (syntax-table)))) (if (or (abbrev-table-get enh-ruby-mode-abbrev-table :parents) (eq enh-ruby-mode-abbrev-table local-abbrev-table)) nil (abbrev-table-put enh-ruby-mode-abbrev-table :parents (list local-abbrev-table)))) (use-local-map enh-ruby-mode-map) (set-syntax-table enh-ruby-mode-syntax-table) (setq local-abbrev-table enh-ruby-mode-abbrev-table) (set (make-local-variable 'comment-column) enh-ruby-comment-column) (set (make-local-variable 'comment-end) "") (set (make-local-variable 'comment-start) "#") (set (make-local-variable 'comment-start-skip) "#+ *") (set (make-local-variable 'erm-buff-num) nil) (set (make-local-variable 'erm-e-w-status) nil) (set (make-local-variable 'erm-full-parse-p) nil) (set (make-local-variable 'indent-line-function) 'enh-ruby-indent-line) (set (make-local-variable 'need-syntax-check-p) nil) (set (make-local-variable 'paragraph-ignore-fill-prefix) t) (set (make-local-variable 'parse-sexp-ignore-comments) t) (set (make-local-variable 'parse-sexp-lookup-properties) t) (set (make-local-variable 'require-final-newline) mode-require-final-newline) (set (make-local-variable 'beginning-of-defun-function) 'enh-ruby-beginning-of-defun) (set (make-local-variable 'end-of-defun-function) 'enh-ruby-end-of-defun) (set (make-local-variable 'show-paren-data-function) 'erm-show-paren-data-function) (set (make-local-variable 'paragraph-start) (concat "$\\|" page-delimiter)) (set (make-local-variable 'paragraph-separate) paragraph-start) (set (make-local-variable 'add-log-current-defun-function) 'enh-ruby-add-log-current-method) (set (make-local-variable 'font-lock-keywords) enh-ruby-font-lock-keywords) (setq font-lock-defaults '((enh-ruby-font-lock-keywords) t)) (setq indent-tabs-mode enh-ruby-indent-tabs-mode) (setq imenu-create-index-function 'enh-ruby-imenu-create-index) (if enh-ruby-add-encoding-comment-on-save (add-hook 'before-save-hook 'enh-ruby-mode-set-encoding nil t)) (add-hook 'change-major-mode-hook 'erm-major-mode-changed nil t) (add-hook 'kill-buffer-hook 'erm-buffer-killed nil t) (abbrev-mode) (erm-reset-buffer))
  (progn (make-local-variable 'delay-mode-hooks) (let ((delay-mode-hooks t)) (prog-mode) (setq major-mode 'enh-ruby-mode) (setq mode-name "EnhRuby") (progn (if (get 'prog-mode 'mode-class) (put 'enh-ruby-mode 'mode-class (get 'prog-mode 'mode-class))) (if (keymap-parent enh-ruby-mode-map) nil (set-keymap-parent enh-ruby-mode-map (current-local-map))) (let ((parent (char-table-parent enh-ruby-mode-syntax-table))) (if (and parent (not (eq parent ...))) nil (set-char-table-parent enh-ruby-mode-syntax-table (syntax-table)))) (if (or (abbrev-table-get enh-ruby-mode-abbrev-table :parents) (eq enh-ruby-mode-abbrev-table local-abbrev-table)) nil (abbrev-table-put enh-ruby-mode-abbrev-table :parents (list local-abbrev-table)))) (use-local-map enh-ruby-mode-map) (set-syntax-table enh-ruby-mode-syntax-table) (setq local-abbrev-table enh-ruby-mode-abbrev-table) (set (make-local-variable 'comment-column) enh-ruby-comment-column) (set (make-local-variable 'comment-end) "") (set (make-local-variable 'comment-start) "#") (set (make-local-variable 'comment-start-skip) "#+ *") (set (make-local-variable 'erm-buff-num) nil) (set (make-local-variable 'erm-e-w-status) nil) (set (make-local-variable 'erm-full-parse-p) nil) (set (make-local-variable 'indent-line-function) 'enh-ruby-indent-line) (set (make-local-variable 'need-syntax-check-p) nil) (set (make-local-variable 'paragraph-ignore-fill-prefix) t) (set (make-local-variable 'parse-sexp-ignore-comments) t) (set (make-local-variable 'parse-sexp-lookup-properties) t) (set (make-local-variable 'require-final-newline) mode-require-final-newline) (set (make-local-variable 'beginning-of-defun-function) 'enh-ruby-beginning-of-defun) (set (make-local-variable 'end-of-defun-function) 'enh-ruby-end-of-defun) (set (make-local-variable 'show-paren-data-function) 'erm-show-paren-data-function) (set (make-local-variable 'paragraph-start) (concat "$\\|" page-delimiter)) (set (make-local-variable 'paragraph-separate) paragraph-start) (set (make-local-variable 'add-log-current-defun-function) 'enh-ruby-add-log-current-method) (set (make-local-variable 'font-lock-keywords) enh-ruby-font-lock-keywords) (setq font-lock-defaults '((enh-ruby-font-lock-keywords) t)) (setq indent-tabs-mode enh-ruby-indent-tabs-mode) (setq imenu-create-index-function 'enh-ruby-imenu-create-index) (if enh-ruby-add-encoding-comment-on-save (add-hook 'before-save-hook 'enh-ruby-mode-set-encoding nil t)) (add-hook 'change-major-mode-hook 'erm-major-mode-changed nil t) (add-hook 'kill-buffer-hook 'erm-buffer-killed nil t) (abbrev-mode) (erm-reset-buffer)))
  enh-ruby-mode()
  set-auto-mode-0(enh-ruby-mode nil)
  set-auto-mode()
  normal-mode(t)
```

**Comment by zenspider** [comment]
I _think_ this is fixed in 28f30c05... but I'm leaving this up for a bit.

**Comment by zenspider** [comment]
I feel like I still see this periodically, but less that I used to... keeping it open for a while longer.

**Comment by zenspider** [comment]
I think I fixed this with 7e76d75 ... I haven't had problems since. 

The above was fixed whenever I was rebasing lots of branches via magit and resolving conflicts via ediff. It often would blow up as I was finishing fixing a conflict and saving it back to the original buffer. 

Since the commit above, I haven't had a problem since.
